phoenix-core/src/main/java/org/apache/hadoop/hbase/regionserver/IndexHalfStoreFileReader.java

```diff
@@ -89,7 +92,12 @@ public IndexHalfStoreFileReader(final FileSystem fs, final Path p, final CacheCo
             final byte[][] viewConstants, final RegionInfo regionInfo,
             byte[] regionStartKeyInHFile, byte[] splitKey, boolean primaryReplicaStoreFile,
             AtomicInteger refCount, RegionInfo currentRegion) throws IOException {
-        super(fs, p, in, size, cacheConf, primaryReplicaStoreFile, refCount, conf);
+        super(new ReaderContext(p, in, size, new HFileSystem(fs), primaryReplicaStoreFile,
+            ReaderType.STREAM),
+            new HFileInfo(new ReaderContext(p, in, size, new HFileSystem(fs),
```

Comment:
Looking at HBase code, it looks like we shouldn't leak any copied filesystem objects here.

We had a lot of perf issues with STREAM reads where they ended up being slower than PREAD. It might be safer to start with PREAD and return to this specifically when we're doing perf testing on this code.